File: Courzelo_Angular/src/app/models/Course/Course.ts
export class Course {
    idCourse?: string;
    title?: string;
    content?: string;
    datecomment?: Date;

    constructor(idCourse?: string, title?: string,content?:string, datecomment?: Date) {
        this.idCourse = idCourse;
        this.title = title;
        this.datecomment = datecomment;
      }
}

File: Courzelo-Base/src/main/java/tn/esprit/devflow/courzelo/entity/Course.java
package tn.esprit.devflow.courzelo.entity;

import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.Size;
import java.time.LocalDate;
import java.util.List;

@Data
@Document
@FieldDefaults(level= AccessLevel.PRIVATE)

public class Course {
    @Id
    String idCourse;
    @Size(max = 55)
     String title;
    String content;
    LocalDate datecomment;
    @DBRef
    List<Lesson> lessons;
    @DBRef
    List<Quiz> quizzes;

}
